src/Parser.js

```diff
@@ -44,7 +44,7 @@ export function parse (value, root, rule, rules, rulesets, points, declarations)
 				break
 			// \t \n \s
 			case 9: case 10: case 32:
-				characters += whitespace(previous)
+				characters += whitespace(previous, strlen(characters))
```

Comment:
This was introduced to pass this test:
https://github.com/thysultan/stylis.js/blob/31d35e159f833f143eb689c2fe34cb8557cbb088/test/Middleware.js#L19
without it, we were getting double whitespace before `:global()` in things like `width:10px; :global()`. The one whitespace came from the `whitespace` and the second one was received from the `ruleset` when props were joined with whitespace, here:
https://github.com/thysultan/stylis.js/blob/31d35e159f833f143eb689c2fe34cb8557cbb088/src/Parser.js#L135

Reply:
The new test doesn't test this though?

Reply:
I've first introduced this change to the `whitespace`:
```diff
- return token(type) > 2 || token(type === 38 && character === 58 ? type : character) > 2 ? '' : ' '
+ return character !== 58 && (token(type) > 2 || token(character) > 2) ? '' : ' '
```

This has removed the dependency on `&` character - to fix the added test because as shown we need to preserve the whitespace for other selectors followed by pseudo as well. 

This has broken the mentioned `namespace` test because it resulted in double whitespace in the output. Besides the double whitespace (non-ideal output) I couldn't yet find if it actually broke anything semantic-wise, but I've decided to introduce this `!len || ` part to avoid this situation.